packages/fuels-signers/src/provider.rs

```diff
@@ -33,6 +36,25 @@ pub struct TransactionCost {
     pub total_fee: u64,
 }
 
+#[derive(Debug)]
+// ANCHOR: time_parameters
+pub struct TimeParameters {
+    // The time to set on the first block
+    pub start_time: u64,
```

Comment:
1. [`Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html) is an opaque measurement, you can only [compare them](https://doc.rust-lang.org/std/time/struct.Instant.html#impl-PartialOrd%3CInstant%3E-for-Instant) to each other or check [time elapsed](https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed) since then. Thus, there's no way to create an `Instant` representing some time point in the past.
2. If you can't compare `Instant`s to fixed values, how will `#[test] can_set_custom_block_time()` work? Is that desirable/OK?
3. If so, can `TimeParameters.block_time_interval` be a [`Duration`](https://doc.rust-lang.org/std/time/struct.Duration.html) (which can be added/subtracted to `Instant`s)?

Reply:
:O didn't know that. Bummer. But I'm definitely for using something that unambiguously tells when something happened. On mobile, not sure if the stdlib has something other than Instant... Still can't believe you can't instantiate a specific point in time.

Reply:
Yup, we need to use something like `chrono` (or `time` for that matter) to construct `DateTime`s or `Timestamp`s. Maybe worthwhile to add it now?